src/main/java/org/apache/sling/models/impl/ModelAdapterFactory.java

```diff
@@ -265,7 +265,11 @@ private void clearDisposalCallbackRegistryQueue() {
     public <AdapterType> AdapterType getAdapter(Object adaptable, Class<AdapterType> type) {
         Result<AdapterType> result = internalCreateModel(adaptable, type);
         if (!result.wasSuccessful()) {
-            log.warn("Could not adapt to model", result.getThrowable());
+            if (result == Result.POST_CONSTRUCT_PREVENTED_MODEL_CONSTRUCTION) {
+                log.warn("Could not adapt to model as PostConstruct method returned false"); // do no construct runtime exception in this case
```

Comment:
should this be converted to debug message? if the postconstruct returned false, this is by intention and not something to be warned?

Reply:
good point, lowered the level to debug in https://github.com/apache/sling-org-apache-sling-models-impl/pull/28/commits/e3afef7b4b175a01ebbf7f7c5bcf0b8607c541bc.